Source/WebCore/loader/cache/CachedCSSStyleSheet.cpp

```diff
@@ -139,11 +139,18 @@ bool CachedCSSStyleSheet::mimeTypeAllowedByNosniff() const
     return parseContentTypeOptionsHeader(response().httpHeaderField(HTTPHeaderName::XContentTypeOptions)) != ContentTypeOptionsDisposition::Nosniff || equalLettersIgnoringASCIICase(responseMIMEType(), "text/css"_s);
 }
 
-bool CachedCSSStyleSheet::canUseSheet(MIMETypeCheckHint mimeTypeCheckHint, bool* hasValidMIMEType) const
+bool CachedCSSStyleSheet::canUseSheet(MIMETypeCheckHint mimeTypeCheckHint, bool* hasValidMIMEType, bool* hasHTTPStatusOK) const
 {
     if (errorOccurred())
         return false;
 
+    // https://html.spec.whatwg.org/#fetching-and-processing-a-resource-from-a-link-element:processresponseconsumebody
+    if (response().httpStatusCode() && !response().isSuccessful()) {
```

Comment:
I find the `response().httpStatusCode()` check here confusing given that we're not using the status code explicitly in the second condition. Would a `!response().isNull()` work? If so, I suggest we switch to that.

Reply:
Thanks — it does seems like `! response.inNull()` would work, but I'll test it locally to make sure, and then push an update

Reply:
> I find the `response().httpStatusCode()` check here confusing given that we're not using the status code explicitly in the second condition. Would a `!response().isNull()` work? If so, I suggest we switch to that.

OK, it seems `!response().isNull()` won’t actually work, because even for `file:` URLs, `response().isNull()` is not true — so `!response().isNull() && !response().isSuccessful()` becomes effectively the same thing as just checking `!response().isSuccessful()`, which for `file:` URLs will always be true.

So I’ve switched the code back to using `response().httpStatusCode() && !response().isSuccessful()`, but also added a comment to clarify what it’s doing and why (that `response().httpStatusCode()` is essentially all about dealing with the `file:` URL case).

I tried to make it even more clear with an explicit `response().httpStatusCode() != 0` — but the style linter doesn’t like that (message: _“Tests for true/false, null/non-null, and zero/non-zero should all be done without equality comparisons”_), so I rolled it back to just being `response().httpStatusCode() && !response().isSuccessful()`.

Reply:
Could we use `response().url().protocolIsInHTTPFamily() && !response().isSuccessful()` then?

Reply:
> Could we use `response().url().protocolIsInHTTPFamily() && !response().isSuccessful()` then?

Aha, didn’t know about that one — will try it right now